apis/networking/v1beta1/apiproduct_types.go

```diff
@@ -106,8 +106,11 @@ type APIProductSpec struct {
 	// platform, short-names can also be used instead of a FQDN (i.e. has no
 	// dots in the name). In such a scenario, the FQDN of the host would be
 	// derived based on the underlying platform.
-	Hosts          []string          `json:"hosts"`
-	SecurityScheme []*SecurityScheme `json:"securityScheme"`
+	Hosts []string `json:"hosts"`
+
+	// Configure authentication mechanisms
+	// +optional
```

Comment:
Is this needed for controller-gen to make it optional? I usually rely only on `omitempty` and it does the trick, but I'd love to learn about any other extra advantage.

Reply:
I have experienced that too. That's confusing as the [kubebulder doc](https://book.kubebuilder.io/reference/markers/crd-validation.html) says it should be the `'+optional` marker.

To make it explicit and easy to read, I always add it for optional fields